pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -343,10 +343,17 @@ func getFirstContainerWithSourceVolume(containers []corev1.Container) (string, e
 }
 
 // getSyncFolder returns the folder that we need to sync the source files to
-// If there's exactly one project defined in the devfile, return `/projects/<projectName`
+// If there's exactly one project defined in the devfile, and clonePath isn't set return `/projects/<projectName>`
+// If there's exactly one project, and clonePath is set, return `/projects/<clonePath>`
 // Otherwise (zero projects or many), return `/projects`
 func getSyncFolder(projects []versionsCommon.DevfileProject) string {
 	if len(projects) == 1 {
+		project := projects[0]
+		// If the clonepath is set to a value, set it to be the sync folder
+		// As some devfiles rely on the code being synced to the folder in the clonepath
+		if project.ClonePath != nil {
+			return filepath.ToSlash(filepath.Join(kclient.OdoSourceVolumeMount, *project.ClonePath))
```

Comment:
The  should be a check to prevent escaping `/projects` using `..`
see: https://github.com/redhat-developer/devfile/blob/master/docs/devfile.md#clonepath

Reply:
Hmm, good catch. 

How should we handle that? Che throws an error if the clonePath starts with either `/` or `..` (https://github.com/eclipse/che/blob/420d4ea34cf50e5835027016211f16b02eba8a54/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/convert/ProjectConverter.java#L109). We should probably do the same.